src/DynamicExpresso.Core/Parsing/Parser.cs

```diff
@@ -2090,23 +2115,39 @@ private MethodData FindBinaryOperator(string operatorName, Expression left, Expr
 
 		private static Expression GenerateStringConcat(Expression left, Expression right)
 		{
-			var concatMethod = typeof(string).GetMethod("Concat", new[] { typeof(object), typeof(object) });
-			if (concatMethod == null)
-				throw new Exception("String concat method not found");
-
-			var rightObj =
-				right.Type.IsValueType
-				? Expression.ConvertChecked(right, typeof(object))
-				: right;
-			var leftObj =
-				left.Type.IsValueType
-				? Expression.ConvertChecked(left, typeof(object))
-				: left;
+			var leftObj = GenerateStringConcatOperand(left);
+			var rightObj = GenerateStringConcatOperand(right);
 
 			return Expression.Call(
-					null,
-					concatMethod,
-					new[] { leftObj, rightObj });
+				null,
+				_concatMethod,
+				new[] { leftObj, rightObj });
+		}
+
+		private static Expression ToStringOrNull(Expression expression)
+		{
+			var nullConstant = Expression.Constant(null);
+			Expression condition = Expression.Equal(
+				expression,
+				nullConstant);
+
+			return Expression.Condition(
+				condition,
+				Expression.ConvertChecked(nullConstant, typeof(String)),
+				Expression.Call(expression, _toStringMethod));
+		}
+
+		private static Expression GenerateStringConcatOperand(Expression expression)
+		{
+			if (expression is ConstantExpression constantExpression &&
```

Comment:
Now that you have `ToStringOrNull` function, I suspect that this `if` is not more useful. What happen if you remove it?

Reply:
Hello @davideicardi 😄 You are right. The `if` is not more required. The tests work fine without it